core/unit_test/TestGraph.hpp

```diff
@@ -349,14 +340,207 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), zero_work_reduce) {
 #if defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOS_ENABLE_CUDA_UVM) && \
     (defined(KOKKOS_ARCH_KEPLER) || defined(KOKKOS_ARCH_MAXWELL))
   if constexpr (std::is_same_v<TEST_EXECSPACE, Kokkos::Cuda>) Kokkos::fence();
-#endif
-#ifdef KOKKOS_ENABLE_HPX  // FIXME_HPX graph.submit() isn't properly enqueued
-  if constexpr (std::is_same_v<TEST_EXECSPACE, Kokkos::Experimental::HPX>)
-    Kokkos::fence();
 #endif
   graph.submit();
   Kokkos::deep_copy(ex, count_host, count);
   ex.fence();
   ASSERT_EQ(count_host(), 0);
 }
+
+// Ensure that an empty graph can be submitted.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), empty_graph) {
+  auto graph = Kokkos::Experimental::create_graph(ex, [](auto) {});
+  graph.instantiate();
+  graph.submit(ex);
+  ex.fence();
+}
+
+template <typename ViewType, size_t TargetIndex, size_t NumIndices = 0>
+struct FetchValuesAndContribute {
+  static_assert(std::is_same_v<typename ViewType::value_type,
+                               typename ViewType::non_const_value_type>);
+
+  ViewType data;
+  typename ViewType::value_type value;
+  Kokkos::Array<size_t, NumIndices> indices{};
+
+  FetchValuesAndContribute(ViewType data_,
+                           std::integral_constant<size_t, TargetIndex>,
+                           typename ViewType::value_type value_)
+      : data(std::move(data_)), value(value_) {}
+
+  FetchValuesAndContribute(ViewType data_,
+                           Kokkos::Array<size_t, NumIndices> indices_,
+                           std::integral_constant<size_t, TargetIndex>,
+                           typename ViewType::value_type value_)
+      : data(std::move(data_)), value(value_), indices(std::move(indices_)) {}
+
+  template <typename T>
+  KOKKOS_FUNCTION void operator()(const T) const {
+    if constexpr (NumIndices > 0) {
+      for (size_t index = 0; index < indices.size(); ++index)
+        data(TargetIndex) += data(indices[index]);
```

Comment:
I am tempted to tell you to can that `Kokkos::Array` and use an old fashioned C array instead so you can write a range-based for loop instead
```C++
for (auto index : indices) {
  data(TargetIndex) += data(index);
}
```
which I find more elegant in this case.  I suspect that this way we might not even need the fancy deduction guide.

Reply:
Surprise surprise... I cannot, because if `NumIndices` is zero, you get:
```bash
/workspaces/kokkos/core/unit_test/TestGraph.hpp:367:18: error: zero size arrays are an extension [-Werror,-Wzero-length-array]
  size_t indices[NumIndices];
```

Reply:
@dalg24 Do we wait for #7293 or not ?

Reply:
Not necessarily no

Reply:
OK, since I left a `FIXME`-like comment, I'll close this thread.